src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -176,6 +168,32 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
 
   }
 
+  def getNIHStatusResponse(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]],
+    loginRequired: Boolean, profile: Profile): Future[PerRequestMessage] = {
+    val dbGapUrl = UserService.groupUrl(FireCloudConfig.Rawls.dbGapAuthorizedUsersGroup)
+    val isDbGapAuthorizedRequest = Get(dbGapUrl)
+
+    pipeline(isDbGapAuthorizedRequest) map { response: HttpResponse =>
+      val authorized: Boolean = response.status match {
+        case x if x == OK => true
+        case _ => false
+      }
+      RequestComplete(OK,
+        NIHStatus(
+          loginRequired,
+          linkedNihUsername = profile.linkedNihUsername,
+          isDbgapAuthorized = Some(authorized),
+          lastLinkTime = Some(profile.lastLinkTime.getOrElse(0L)),
+          linkExpireTime = Some(profile.linkExpireTime.getOrElse(0L)),
```

Comment:
linkExpireTime is wrong here. Note that above, linkExpireSeconds is a var. If the previous code determines that the user's link expire time should be reset, it persists the new expire time to Thurloe and updates the var. Here, you should be using the updated var, not the original value from profile.